File: 프로그래머스/lv2/42584. 주식가격/주식가격.java
import java.util.*;
class Solution {
   public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        Deque<Integer> deque = new ArrayDeque<>();
        deque.push(0);
        for(int i=1;i<prices.length;i++){
            Deque<Integer> newDeque = new ArrayDeque<>();
            while(!deque.isEmpty()){
                int x=deque.pop();
                answer[x]++;
                if(prices[x]<=prices[i]) newDeque.push(x);
            }
            newDeque.push(i);
            deque=newDeque;
        }
        return answer;
    }
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = [0]*len(prices)
    for i in range(0, len(prices)-1):
        j = i+1
        while j< len(prices):
            if prices[i]>prices[j]:
                answer[i]=j-i
                break
            j+=1
        if j==len(prices):
            answer[i]=len(prices)-i-1
    return answer